### PR TITLE
remove volta from CI

### PR DESCRIPTION
**Pull request by mansona**
This PR attempts to just use the standard `actions/setup-node` infra to setup node because the volta infra keeps crashing on us

### CODE REVIEW

File: .github/actions/setup/action.yml [hunk 1]
```diff
@@ -8,19 +8,14 @@ inputs:
...
-    - name: Get pnpm cache directory path
-      id: pnpm-cache-dir-path
-      run: echo "STORE_PATH=$(pnpm store path)" >> $GITHUB_OUTPUT
-      shell: bash
-    - uses: actions/cache@v3
-      id: pnpm-cache
+    - uses: pnpm/action-setup@v2
+      name: Install pnpm
+      with:
+        version: 7
```
Comment (void-mAlex) [inline .github/actions/setup/action.yml:14]: don't we want pnpm version 8?

Comment (mansona) [reply .github/actions/setup/action.yml:14]: lol yes, but also I just noticed that our volta was pinning us to before the `resolution-mode setting is changed to highest` shenanigans so we'll see if anything breaks 🙃

**Review by NullVoxPopuli — APPROVED** [review]

File: .npmrc [hunk 2]
```diff
@@ -1 +1,2 @@
-auto-install-peers=false
\ No newline at end of file
+auto-install-peers=false
```
Comment (NullVoxPopuli) [inline .npmrc:1]: we probably also want: https://github.com/ember-cli/ember-cli/blob/master/blueprints/addon/files/.npmrc#L8

can be separate, ofc

**Comment by void-mAlex** [comment]
seems like the one test that is failing is related directly to volta shenanigans ... we may need to see how that translates in a world without volta
https://github.com/void-mAlex/embroider/blob/main/tests/scenarios/shared-internals-test.ts#L44

File: tests/scenarios/shared-internals-test.ts [hunk 3]
```diff
@@ -39,14 +39,7 @@ Scenarios.fromProject(() => new Project('shared-internals-tests'))
...
 
-        // if we just try to invoke node directly in a child process, our own
-        // volta settings dominate over the test app's
-        let tryit = await app.execute('volta which node');
-        if (tryit.exitCode !== 0) {
-          throw new Error('unable to locate our node version');
-        }
-        let nodebin = tryit.output.trim();
-        let result = await app.execute(`${nodebin} ./node_modules/qunit/bin/qunit.js ./test.js`);
+        let result = await app.execute(`node ./node_modules/qunit/bin/qunit.js ./test.js`);
```
Comment (void-mAlex) [inline tests/scenarios/shared-internals-test.ts:42]: the above test is names testing on node 12 ... from qunit

Comment (mansona) [reply tests/scenarios/shared-internals-test.ts:42]: I added testing this with node 12 as a separate job, I was going to remove it from the main matrix but I thought it wouldn't help 🤷

**Comment by ef4** [comment]
Volta maintainers discussing a fix: https://github.com/volta-cli/volta/issues/1523#issuecomment-1732187410

**Comment by mansona** [comment]
I'm submitting this PR for consideration again. I have implemented the pnpm node version management for the tests that need it but I have not implemented it across the board.

My reasoning for this is that we specifically have tests that use `--no-lockfile` because we want to track when our CI is failing because of changes in our dependencies. With volta and pnpm's `use-node-version` implementation you have to pin an exact version of node which does not seem in the spirit of `--no-lockfile`. This PR also identified the problem that I describe in #1649 because it also didn't pin an exact version of pnpm

I will add this PR to the meeting agenda for discussion 👍

**Review by NullVoxPopuli — APPROVED** [review]

**Review by ef4 — APPROVED** [review]

**Pull request merged**